### ISSUE TITLE
[UI/UX] Ability to remap controller buttons

### ISSUE THREAD

**Issue by wonx** [rommapp/romm#1250]
Hi,

I love that RomM embeds emulatorjs to play games in the browser. However, I noticed that the default controller mapping switches A and B, and Y and X in my gamepad (a [8BitDo Pro 2](https://www.8bitdo.com/pro2/)). I can reconfigure it in the "Control settings" section when the game is running, but I believe these changes only reside in the browser window and only for that specific system.

I'd like a way to be able to permanently remap controller buttons. I found this in emulatorjs documentation https://emulatorjs.org/docs4devs/control-mapping , but I don't even know where to start making those changes.

Is this currently possible?

**Comment by Gibbz** [comment]
Would it be easier to start with having the mappings per device in a config file? At least it would work until a UI could be implemented.

**Comment by gantoine** [comment]
@Gibbz Yeah I think that would work! This is backlogged till someone gets to it, but its a good option for whoever picks it up.

**Comment by deluxe2** [comment]
I would like to work on this feature, if nobody else is working on it. I have a german keyboard and playing with swapped y and z is kinda awkward :D

**Comment by zurdi15** [comment]
> I would like to work on this feature, if nobody else is working on it. I have a german keyboard and playing with swapped y and z is kinda awkward :D

Of course bud! go for it, nobody is working on this atm

**Comment by gantoine** [comment]
@deluxe2 whatever you end up building a) i don't think it needs to save on the server (ie client side only/localstorage is fine) and b) each user should be able to set their own values for remapping. and ideally whatever you build could be extended to pass more options to emulatorjs!

**Comment by gantoine** [comment]
This will be released in v4.2!

**Comment by Kamikaze01git** [comment]
yeah... looking forward to it 👍

**Comment by kzieres61** [comment]
I'm here to check on something. I've got version 4.2, and I can only remap controllers when a game is launched. Also, is it possible to KEEP the mappings for a second player? I have to keep mapping them every time I launch a new game.

**Comment by gantoine** [comment]
> I can only remap controllers when a game is launched
> Also, is it possible to KEEP the mappings for a second player

https://docs.romm.app/latest/Getting-Started/Configuration-File/#controls
